Features/Transfer/Sources/Scenes/ConfirmTransferScene.swift

```diff
@@ -20,9 +20,8 @@ public struct ConfirmTransferScene: View {
             Spacer()
             StateButton(
                 text: model.buttonTitle,
-                viewState: model.state,
+                variant: .primary(model.state, disabledRule: model.shouldDisableButton),
```

Comment:
```suggestion
                variant: .primary(model.state, disabledRule: model.shouldDisableButton),
```

`disabledRule` vs `shouldDisableButton`

need consistent naming as `disableRule` and `shouldDisableButton` or `disableRule`

or ideally: `.primary(model.state, isEnabled: model.isEnabledButton)`.